tests/Unit/Test/Normalizer/TestNormalizerTest.php

```diff
@@ -95,21 +97,107 @@ public function testNormalize(): void
             ->expects($this->once())
             ->method('getJsonTest')
             ->with($test)
-            ->willReturn('{"identifier": "test_id"}');
-
-        $this->assertEquals(
-            [
-                'type' => ['document type'],
-                'item_uris' => [
-                    'item://1',
-                    'item://2'
+            ->willReturn($testJson);
+
+        $this->assertEquals($expectedNormalizedBody, $this->sut->normalize($test)->getBody());
+    }
+
+    public function testDataProvider(): array
+    {
+        $qtiTestStructureJson = '{"qti-type":"assessmentTest","identifier":"testIdentifier","title":"title","toolVersion":"beta","testParts":[{"qti-type":"testPart","identifier":"cluster-id-stage1","navigationMode":0,"submissionMode":0,"preConditions":[],"branchRules":[],"itemSessionControl":{"maxAttempts":0,"qti-type":"itemSessionControl"},"assessmentSections":[{"qti-type":"assessmentSection","title":"Section","identifier":"assessmentSection-0","preConditions":[],"branchRules":[],"index":0,"sectionParts":[{"href":"https://itemUri1","label":"item-1","qti-type":"assessmentItemRef","categories":["cluster-id-stage1","x-tao-option-reviewScreen"],"identifier":"item1","timeLimits":{"qti-type":"timeLimits","maxTime":60}}],"timeLimits":{"qti-type":"timeLimits","maxTime":60}}]},{"qti-type":"testPart","identifier":"cluster-id-stage1-Unit11","navigationMode":0,"submissionMode":0,"preConditions":[],"branchRules":[],"assessmentSections":[{"qti-type":"assessmentSection","title":"Section","visible":true,"identifier":"assessmentSection-1","required":true,"preConditions":[],"branchRules":[],"index":0,"itemSessionControl":{"maxAttempts":0,"qti-type":"itemSessionControl"},"categories":[],"isSubsection":false,"sectionParts":[{"href":"https://itemUri0","label":"item-0","qti-type":"assessmentItemRef","categories":["cluster-id-stage1","x-tao-option-reviewScreen"],"identifier":"item-0","timeLimits":{"qti-type":"timeLimits","maxTime":60}}],"timeLimits":{"qti-type":"timeLimits","maxTime":60}}]}],"timeLimits":{"maxTime":3600,"qti-type":"timeLimits"}}'; // phpcs:ignore
```

Comment:
Thanks for the test! Some points:

- Do you think we really need all this JSON content in the test case? I think the identifiers, branch rules, and other attributes we do not really use can be removed, so the test is more atomic and easy to maintain.
- Can you please indent the JSON for easy readability?

Thanks!

Reply:
moved out to a file